Core/include/Acts/EventData/detail/coordinate_transformations.hpp

```diff
@@ -126,9 +126,29 @@ struct coordinate_transformation {
   static double parameters2charge(const ParVector_t& pars) {
     return (pars(Acts::eQOP) > 0) ? 1. : -1.;
   }
+
+  /// @brief Transforms a bound parameter vector into the free equivalent
+  ///
+  /// @param [in] gtcx Geometry context
+  /// @param [in] parameters Bound parameter vector
+  /// @param [in] surface Surface related to @p parameters
+  ///
+  /// @return FreeVector representation of @p parameters
+  static FreeVector boundParameters2freeParameters(
+      const GeometryContext& gtcx, const BoundVector& parameters,
+      const Surface& surface) {
+    FreeVector result;
+    result.template segment<3>(eFreePos0) =
+        parameters2globalPosition(gtcx, parameters, surface);
+    result[eFreeTime] = parameters[eBoundTime];
+    result.template segment<3>(eFreeDir0) =
+        parameters2globalMomentum(parameters).normalized();
+    result[eFreeQOverP] = parameters[eBoundQOverP];
+    return result;
+  }
```

Comment:
This is extracted from `BoundParameters`, yes?

Reply:
What do you mean with `extracted`? The methods called in here are defined in this file.

Reply:
The logic existed before somewhere, right?. I believe `parameters2globalPosition` and `parameters2globalMomentum` were called from the `BoundParameters` constructor, is that correct?

Reply:
Ah yes, there you convert it such that you have the local parameters in the `ParameterSet` and the global ones for easy access as members in the `SingleTrackParameters`